pkg/virt-controller/services/template.go

```diff
@@ -499,6 +499,25 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 			}
 			sidecarVolumes = append(sidecarVolumes, vol)
 		}
+		if requestedHookSidecar.PVC != nil {
+			volumeSource := k8sv1.VolumeSource{
+				PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+					ClaimName: requestedHookSidecar.PVC.Name,
+				},
+			}
+			vol := k8sv1.Volume{
+				Name:         requestedHookSidecar.PVC.Name,
+				VolumeSource: volumeSource,
+			}
+			sidecarVolumes = append(sidecarVolumes, vol)
+			if requestedHookSidecar.PVC.SharedComputePath != "" {
+				containers[0].VolumeMounts = append(containers[0].VolumeMounts,
```

Comment:
WDYT about using `compute.VolumeMounts` instead of `containers[0].VolumeMounts` because, IIUC, we're appending to `compute` container's volume mounts here and the variable `compute` is a pointer: https://github.com/kubevirt/kubevirt/blob/d7371feb56fcb138806b1b6e5b6ab154e900a2e0/pkg/virt-controller/services/template.go#L417

Reply:
The containers aren't a pointers, so, I don't think that modifying the compute object will be reflected in the containers array. AFAIU, they are full copies